maggy/maggy/api/routes_pr_review.py

```
@@ -0,0 +1,109 @@
+@router.post("/run")
+async def run(request: Request, body: RunRequest, x_api_key: str | None = Header(None)) -> dict:
+    """Run the council on a PR. dry_run (default from config) skips the GitHub write."""
+    check_auth(request, x_api_key)
+    cfg = request.app.state.cfg
+    token = _resolve_token(cfg, body.github_token)
+    if not token:
+        raise HTTPException(400, "No GitHub token — set review.github_token, pass github_token, or export GITHUB_TOKEN")
+    try:
+        from maggy.review.pipeline import run_review
+    except ImportError:
+        raise HTTPException(503, "Reviewer not installed — run: pip install maggy-harness[review]")
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)